src/lib/react/react.js

```diff
@@ -0,0 +1,60 @@
+export const React = (() => {
+  let globalHooksMap = new Map();
+  let globalHooks = null;
+  let globalHookIndex = 0;
+
+  // Reset React
+  const __reset = () => {
+    globalHooksMap = new Map();
+    globalHooks = null;
+    globalHookIndex = 0;
+  };
+
+  // Reset hooks before rendering
+  const __prepareForRender = (Component) => {
+    if (!globalHooksMap.has(Component)) {
+      globalHooksMap.set(Component, []);
+    }
+
+    globalHooks = globalHooksMap.get(Component);
+    globalHookIndex = 0;
+  };
+
+  const useState = (initialValue) => {
+    if (globalHooks === null) {
+      throw new Error("NOT_READY");
+    }
+
+    const currentHooks = globalHooks;
+    const currentHookIndex = globalHookIndex;
+
+    const setCurrentHook = (value) => {
+      currentHooks[currentHookIndex] = value;
+    };
+
+    if (currentHooks.at(currentHookIndex) === undefined) {
+      setCurrentHook(initialValue);
+    }
+
+    const setState = (state) => {
+      if (typeof state === "function") {
+        const newValue = state(currentHooks.at(currentHookIndex));
+        setCurrentHook(newValue);
+        return;
+      }
+
+      setCurrentHook(state);
+    };
```

Comment:
p2: 이쪽 부분 dom 비교 업데이트가 들어가면 변경될 부분이긴 하지만, setState 되었을 때 rerender 되는 로직이 들어가면 좋을 것 같아요. 아래와 같은 느낌으로요! 

```suggestion
    const setState = (state) => {
      if (typeof state === "function") {
        const newValue = state(currentHooks.at(currentHookIndex));
        setCurrentHook(newValue);
        return;
      }

      setCurrentHook(state);
    
        __reRender();
    };
```